plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushMetadataCacheProcedure.java

```diff
@@ -109,22 +113,28 @@ public void flushMetadataCache(
 
     private void doFlushMetadataCache(Optional<String> schemaName, Optional<String> tableName, List<String> partitionColumns, List<String> partitionValues)
     {
-        CachingHiveMetastore cachingHiveMetastore = this.cachingHiveMetastore
-                .orElseThrow(() -> new TrinoException(HiveErrorCode.HIVE_METASTORE_ERROR, "Cannot flush, metastore cache is not enabled"));
+        if (cachingHiveMetastore.isEmpty() && glueCache.isEmpty()) {
+            // TODO this currently does not work. CachingHiveMetastore is always bound for metastores other than Glue, even when caching is disabled,
```

Comment:
Using an optional binder on a "flush" interface may be a way to avoid this problem.  We would just need to do bind the interface when shared cache is enabled or glue cache is enabled.

Reply:
i actually don't think we can restore this logic now. 
people could become dependent on the fact that they can call flush any time they want. it would be a breaking change.

i didn't want to deal with this within this PR, hence only a TODO added